src/pages/iou/request/step/IOURequestStepTaxAmountPage.js

```diff
@@ -0,0 +1,141 @@
+import HeaderWithBackButton from '@components/HeaderWithBackButton';
+import ScreenWrapper from '@components/ScreenWrapper';
+import transactionPropTypes from '@components/transactionPropTypes';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import compose from '@libs/compose';
+import * as CurrencyUtils from '@libs/CurrencyUtils';
+import * as IOUUtils from '@libs/IOUUtils';
+import Navigation from '@libs/Navigation/Navigation';
+import MoneyRequestAmountForm from '@pages/iou/steps/MoneyRequestAmountForm';
+import reportPropTypes from '@pages/reportPropTypes';
+import * as IOU from '@userActions/IOU';
+import CONST from '@src/CONST';
+import ROUTES from '@src/ROUTES';
+import IOURequestStepRoutePropTypes from './IOURequestStepRoutePropTypes';
+import withFullTransactionOrNotFound from './withFullTransactionOrNotFound';
+import withWritableReportOrNotFound from './withWritableReportOrNotFound';
+
+const propTypes = {
+    /** Navigation route context info provided by react navigation */
+    route: IOURequestStepRoutePropTypes.isRequired,
+
+    /* Onyx Props */
+    /** The report that the transaction belongs to */
+    report: reportPropTypes,
+
+    /** The transaction object being modified in Onyx */
+    transaction: transactionPropTypes,
+};
+
+const defaultProps = {
+    report: {},
+    transaction: {},
+};
+
+function IOURequestStepTaxAmountPage({
+    route: {
+        params: {iouType, reportID, transactionID, backTo, currency: selectedCurrency},
+    },
+    transaction,
+    transaction: {currency: originalCurrency},
+    report,
+}) {
+    const {translate} = useLocalize();
+    const styles = useThemeStyles();
+    const textInput = useRef(null);
+    const isEditing = Navigation.getActiveRoute().includes('taxAmount');
+
+    const currency = selectedCurrency || originalCurrency;
+
+    const focusTimeoutRef = useRef(null);
+
+    useFocusEffect(
+        useCallback(() => {
+            focusTimeoutRef.current = setTimeout(() => textInput.current && textInput.current.focus(), CONST.ANIMATED_TRANSITION);
+            return () => {
+                if (!focusTimeoutRef.current) {
+                    return;
+                }
+                clearTimeout(focusTimeoutRef.current);
+            };
+        }, []),
+    );
+
+    const navigateBack = () => {
+        Navigation.goBack(isEditing ? ROUTES.MONEY_REQUEST_CONFIRMATION.getRoute(iouType, reportID) : ROUTES.HOME);
+    };
+
+    const navigateToCurrencySelectionPage = () => {
+        // If the money request being created is a distance request, don't allow the user to choose the currency.
+        // Only USD is allowed for distance requests.
+        // Remove query from the route and encode it.
+        Navigation.navigate(ROUTES.MONEY_REQUEST_STEP_CURRENCY.getRoute(iouType, transactionID, reportID, backTo ? 'confirm' : '', Navigation.getActiveRouteWithoutParams()));
```

Comment:
confirm can be translated?

Reply:
Not sure we need to translate here, although I lifted this piece from the current refactor, just to maintain the flow.